### ISSUE TITLE
Port v0 Invariant Interface

### ISSUE THREAD

**Issue by wggster** [DylanLukes/renkon#5]

**Comment by wggster** [comment]
- [ ] Tear out the sketch code
- [ ] Implement a simple batch interface (for testing)
- [ ] Identify the minimal model invariant interface in v0 code

**Comment by DylanLukes** [comment]
v0 interface's `infer` method doesn't translate well, blocked by/related to #6, moving there to avoid carrying over technical debt before implementing batch interface (promoting that item to a new issue).

**Comment by DylanLukes** [comment]
First-pass finished. May need to change in the future, but for now should be stable enough.

For `StatTrait`, `fit`, `test_inyling`.

For `PropTrait`, `satisfy`, `test_satisfied`. 

What's missing or will evolve is some notion of "confidence"

**Comment by DylanLukes** [comment]
`Implementing a simple batch interface (for testing) (6 hr)` was moved to its own task.
